### PR TITLE
Set `FLEET` in env variable

### LINKED ISSUE #5213: Add way to select different fleets for use with the app
### User Story

As a sysadmin(devops is a buzzword) I want the app to be able to switch between different `eth` fleets without having to re-build the whole application. This is necessary to test infrastructure changes and `statusd` version changes on a __staging__ fleet before it's deployed to the fleet used in the public release.

### Description

Right now we have 3 fleets:
* [`eth.test`](https://consul.status.im/ui/do-ams3/nodes?s=eth.test) - Used by me for developing infra.
* [`eth.staging`](https://consul.status.im/ui/do-ams3/nodes?s=eth.staging) - Intended for `statusd` CI and testing new versions.
* [`eth.beta`](https://consul.status.im/ui/do-ams3/nodes?s=eth.beta) - The fleet currently used in the public releases.

Onle one of them is used in the app currently, the `eth.beta`.

#### Expected behavior
In order to test for example a new `statusd` version on `eth.staging` before deploying it to `eth.beta` we need to be able to select one of those in the App before logging in. This should be done in such a way that the node does not use any of the cached nodes.

#### Actual behavior
Currently when you log in you default to using the `boot`, `whisper` and `mail` nodes defined in two files:
* [`status-im/status-react/blob/develop/src/status_im/constants.cljs`](https://github.com/status-im/status-react/blob/develop/src/status_im/constants.cljs#L88)
* [`status-im/status-go/blob/develop/params/cluster.go`](https://github.com/status-im/status-go/blob/develop/params/cluster.go#L37)

This is hard to maintain and far too static.

### Additional Information
There was an issue cut for the `status-go` repo but it might be superfluous: https://github.com/status-im/status-go/issues/1100

I'm not sure what is the best way to do this but my gut is telling me that the best way would be to have a JSON file that `status-react` loads, that can be edited by devs/testers by hand if need be, that is then used to inform `status-go` library used by `status-react` on what to connect to. This way configuration will be just in one palce(`status-react`) and will be editable without the need to rebuild anything.

Something like:
```json
{
   "mainnet": {
       "default": "eth.beta",
       "eth.beta": {
           "boot": {
                "boot-01.do-ams3.eth.beta": "enode://ABCxyz@174.138.105.243:30404",
                "boot-02.do-ams3.eth.beta": "enode://XYZabc@206.189.243.57:30404",
                ...
           },
           "whisper": {
                ...
           },
           "mail": {
                ...
           }
       },
       "eth.staging": {
          ...
       }
   },
   "ropsten": {
      ...
   }
}
```

### PR DESCRIPTION
**Pull request by adambabik**
Allow to specify a fleet in the `env` file and use it in the configuration.

Todo:
- [x] provide Mail Servers for `env.staging`.

Closes #5213

### CODE REVIEW

**Comment by adambabik** [comment]
@mandrigin which `.env` file is used by the CI job trigged by this PR?

**Comment by adambabik** [comment]
`eth.staging` mail servers:
1. `enode://69f72baa7f1722d111a8c9c68c39a31430e9d567695f6108f31ccb6cd8f0adff4991e7fdca8fa770e75bc8a511a87d24690cbc80e008175f40c157d6f6788d48@206.189.240.16:30504`,
2. `enode://e4fc10c1f65c8aed83ac26bc1bfb21a45cc1a8550a58077c8d2de2a0e0cd18e40fd40f7e6f7d02dc6cd06982b014ce88d6e468725ffe2c138e958788d0002a7f@35.239.193.41:30504`

**Comment by mandrigin** [comment]
@adambabik just `.env`

**Review by mandrigin — APPROVED** [review]
LGTM!

**Review by pedropombeiro — APPROVED** [review]
LGTM

**Comment by mandrigin** [comment]
@adambabik maybe later you can make a separate PR for `eth.staging` builds that won't be merged? Just for tests?
After that, you can create one more .env and I can add another nightly or release job to build an app for eth.staging

**Comment by mandrigin** [comment]
@adambabik afaik, staging mailservers are in `src/status_im/constants.cljs`

**Comment by adambabik** [comment]
@mandrigin I think we can use a parametrized build in Jenkins if one needs to test against the `eth.staging`. I already changed the job. My question is if that job is reliable.

**Comment by adambabik** [comment]
@mandrigin I don't see these mail servers in that file. There are different. I would need some help with someone familiar with Clojure to change that, pretty much the task is to remove Mail Servers per network as they all are the same now and instead we should have Mail Servers per fleet.

**Comment by adambabik** [comment]
There is some issue with a parametrized build (https://ci.status.im/job/status-react/job/parametrized_build/761/console) but this Android version should work on "eth.staging": https://i.diawi.com/MeGGkw

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/parametrized_build/761/console">status-react » parametrized_build #761 Console [Jenkins]</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/MeGGkw">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java [hunk 1]
```diff
@@ -242,9 +244,10 @@ private void doStartNode(final String defaultConfig) {
 
             }
             try {
-                Object clusterConfig = customConfig.get("ClusterConfig");
+               JSONObject clusterConfig = (JSONObject) customConfig.get("ClusterConfig");
```
Comment (pedropombeiro) [inline modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java]: Missing one space in indentation

**Comment by mandrigin** [comment]
@adambabik 
> There is some issue with a parametrized build (https://ci.status.im/job/status-react/job/parametrized_build/761/console)

rebasing to the latest `develop` should fix it

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/parametrized_build/761/console">status-react » parametrized_build #761 Console [Jenkins]</a></strong></div></blockquote>

**Comment by adambabik** [comment]
I merged develop and pulled with rebase which was a mistake... I will squash it before merging to develop so all should be fine. I don't want to force push as multiple people committed already.

**Comment by adambabik** [comment]
It's ready to be tested. I build it with `eth.staging` fleet as well: https://ci.status.im/job/status-react/job/parametrized_build/763/

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/parametrized_build/763/">status-react » parametrized_build #763 [Jenkins]</a></strong></div></blockquote>

**Comment by mandrigin** [comment]
@yenda @adambabik I'll take over this one if you don't mind :-)

**Comment by annadanchenko** [comment]
Issue 1. beta fleet nodes are used instead of staging, e.g. `436cc6f674928fdc9a9f7990f2944002b685d1c37f025c1be425185b5b1f0900feaf1ccc2a6130268f9901be4a7d252f37302c8335a2c1a62736e9232691cc3a` is used 

TF session: https://app.testfairy.com/projects/4803622-status/builds/8563803/sessions/4398968123/?accessToken=LDKbiNAO2RRireCbgKrcwPugxeg
Build: https://jenkins.status.im/job/status-react/job/parametrized_build/770/ 

Issue 2 (not sure it's expected): Mailservers list is empty 
screenshot_20180820-172836


<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8563803/sessions/4398968123/?accessToken=LDKbiNAO2RRireCbgKrcwPugxeg">TestFairy: Status 0.9.26 (6461) @2018</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://jenkins.status.im/job/status-react/job/parametrized_build/770/">status-react » parametrized_build #770 [Jenkins]</a></strong></div></blockquote>

File: src/status_im/constants.cljs [hunk 2]
```diff
@@ -83,81 +83,64 @@
...
 (def default-wnodes
-  (assoc default-wnodes-without-custom :custom (:testnet default-wnodes-without-custom)))
+  ;; We use the same set of mailservers for every network now
+  ;; They are only dependent on the selected fleet (test, stage, beta)
+  (let [nodes-for-fleet (get default-wnodes-without-custom config/fleet)]
+    (-> {}
+        (assoc :custom nodes-for-fleet)
+        (assoc :testnet nodes-for-fleet)
+        (assoc :mainnet nodes-for-fleet)
+        (assoc :rinkeby nodes-for-fleet))))
```
Comment (yenda) [inline src/status_im/constants.cljs]: This is equivalent to

```
(let [nodes-for-fleet (get default-wnodes-without-custom config/fleet)]
  {:custom nodes-for-fleet
   :testnet nodes-for-fleet
   :mainnet nodes-for-fleet
   :rinkeby nodes-for-fleet})
```

**Review by yenda — CHANGES_REQUESTED** [review]

**Review by yenda — APPROVED** [review]

File: src/status_im/constants.cljs (hunk 2 above)
Comment (mandrigin) [reply src/status_im/constants.cljs]: yeah, I overcomplicated it 😆

Comment (mandrigin) [reply src/status_im/constants.cljs]: fixed

**Comment by annadanchenko** [comment]
with https://ci.status.im/job/status-react/job/nightly_staging_fleet/5/console build can connect to staging fleet (see mailserver and bootnode from staging nodes), no mailserver from beta fleet. https://app.testfairy.com/projects/4803622-status/builds/8565108/sessions/4399080579/?accessToken=8ZlfCTYroufGrZm9tMa6psSCgjk

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/nightly_staging_fleet/5/console">status-react » nightly_staging_fleet #5 Console [Jenkins]</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8565108/sessions/4399080579/?accessToken=8ZlfCTYroufGrZm9tMa6psSCgjk">TestFairy: Status 0.9.26 (6478) @2018</a></strong></div></blockquote>

**Pull request merged**